test/interactions/dragInteractionTests.ts

```diff
@@ -1,376 +1,410 @@
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        component = new Plottable.Component();
+        component.renderTo(svg);
+
+        dragInteraction = new Plottable.Interactions.Drag();
+      });
+
+      it("calls the onDragStart() callback", () => {
+        let startCallbackCalled = false;
+        let receivedStart: Plottable.Point;
+        let startCallback = (point: Plottable.Point) => {
+          startCallbackCalled = true;
+          receivedStart = point;
+        };
+
+        assert.strictEqual(dragInteraction.onDragStart(startCallback), dragInteraction,
+          "setting onDragStart callback returns the drag interaction");
+        dragInteraction.attachTo(component);
+
+        let target = component.background();
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y);
+        assert.isTrue(startCallbackCalled, "callback was called on beginning drag (mousedown)");
+        assert.deepEqual(receivedStart, startPoint, "was passed the correct point");
+
+        startCallbackCalled = false;
+        receivedStart = null;
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y, 2);
+        assert.isFalse(startCallbackCalled, "callback is not called on right-click");
+
+        startCallbackCalled = false;
+        receivedStart = null;
+        TestMethods.triggerFakeTouchEvent("touchstart", target, [{x: startPoint.x, y: startPoint.y}]);
+        assert.isTrue(startCallbackCalled, "callback was called on beginning drag (touchstart)");
+        assert.deepEqual(receivedStart, startPoint, "was passed the correct point");
+
+        startCallbackCalled = false;
+        TestMethods.triggerFakeMouseEvent("mousedown", target, outsidePointPos.x, outsidePointPos.y);
+        assert.isFalse(startCallbackCalled, "does not trigger callback if drag starts outside the Component (positive) (mousedown)");
+        TestMethods.triggerFakeMouseEvent("mousedown", target, outsidePointNeg.x, outsidePointNeg.y);
+        assert.isFalse(startCallbackCalled, "does not trigger callback if drag starts outside the Component (negative) (mousedown)");
+
+        TestMethods.triggerFakeTouchEvent("touchstart", target, [{x: outsidePointPos.x, y: outsidePointPos.y}]);
+        assert.isFalse(startCallbackCalled, "does not trigger callback if drag starts outside the Component (positive) (touchstart)");
+        TestMethods.triggerFakeTouchEvent("touchstart", target, [{x: outsidePointNeg.x, y: outsidePointNeg.y}]);
+        assert.isFalse(startCallbackCalled, "does not trigger callback if drag starts outside the Component (negative) (touchstart)");
+
+        dragInteraction.offDragStart(startCallback);
+
+        startCallbackCalled = false;
+        TestMethods.triggerFakeMouseEvent("mousedown", target, startPoint.x, startPoint.y);
+        assert.isFalse(startCallbackCalled, "callback was decoupled from the interaction");
```

Comment:
This gets tested here, but also below in the next `it()`. Does it make sense to check it twice? Also this test's name doesn't indicate that it's also testing detaching.